File: backend/orderHandleServices/internal/model/Order.go
package model

import (
	"gorm.io/gorm"
	"time"
)

type Orders struct {
	gorm.Model
	Id             int64          `json:"id" gorm:"primary_key;AUTO_INCREMENT"` // 订单数据Id
	OrderId        string         `json:"order_id" gorm:"unique"`               // 订单号码
	UserId         int64          `json:"user_id"`                              // 用户ID
	GroupId        int64          `json:"group_id"`                             // 群组id
	Email          string         `json:"email"`                                // 用户邮箱地址
	PlanId         int64          `json:"plan_id"`                              // 选择的订阅计划ID
	PlanName       string         `json:"plan_name"`                            // 订阅计划名
	PaymentMethod  string         `json:"payment_method"`                       // 付款方式
	Period         string         `json:"period"`                               // 订阅周期id
	CouponId       int64          `json:"coupon_id"`                            // 如使用优惠券的id
	CouponName     string         `json:"coupon_name"`                          // 如果使用了优惠券这里为优惠券的名称
	Status         int8           `json:"status" gorm:"COMMENT:'1新购2续费3升级'"`    // 1新购2续费3升级
	IsFinished     bool           `json:"is_finished"`                          // 该订单是否完成
	IsSuccess      bool           `json:"is_success"`                           // 是否支付成功
	FailureReason  string         `json:"failure_reason"`                       // 如果支付失败，记录失败原因
	DiscountAmount float64        `json:"discount_amount"`                      // 抵折金额
	Amount         float64        `json:"amount"`                               // 订单金额
	Price          float64        `json:"price"`                                // 这是订单原始的价格
	PaidAt         *time.Time     `json:"paid_at"`                              // 支付时间
	CreatedAt      time.Time      `json:"created_at"`                           // 下单时间
	UpdatedAt      time.Time      `json:"updated_at"`                           // 更新时间
	DeletedAt      gorm.DeletedAt `json:"deleted_at"`                           // 软删除时间
	User           User           `gorm:"foreignKey:UserId;references:Id"`      // 多对一关联，User 表中的用户
	//ActiveOrder    ActiveOrders   `gorm:"foreignKey:OrderId;references:OrderId"` // 外键关联到 Orders 表中的 OrderId
}

func (Orders) TableName() string {
	return "x_orders"
}

File: nxc-frontend/src/api/admin/order.ts
import instance from "@/axios"

type Order = {
    id: number
    user_id: number
    order_id: string
    email: number
    plan_id: number
    group_id: number
    plan_name: string
    period: string
    coupon_id: number
    coupon_name: string
    status: number // 0新购 1续费 2修改订阅
    is_success: boolean
    is_finished: boolean
    price: number
    amount: number
    discount_amount: number
    paid_at: string
    payment_method: string
    created_at: string
    failure_reason: string
}

export const h1 = (page: number, size: number) => {
    try {

    } catch (error) {
        console.error(error)
        return false
    }
}

export const handleGetAllOrders = async (page: number, size: number, searchEmail: string, sort: string) => {
    try {
        let {data} = await instance.get('/api/admin/v1/order', {
            params: {
                page: page,
                size: size,
                search_email: searchEmail,
                sort: sort,
            }
        })
        return data
    } catch (error) {
        console.error(error)
        return false
    }
}

export const handleManualCancelOrderById = async (userId: number, orderId: string) => {
    try {
        let {data} = await instance.delete('/api/admin/v1/order', {
            params: {
                user_id: userId,
                order_id: orderId
            }
        })
        return data
    } catch (error) {
        console.error(error)
        return false
    }
}

export const handleManualPassOrderById = async (userId: number, orderId: string) => {
    try {
        let {data} = await instance.put('/api/admin/v1/order', {
            user_id: userId,
            order_id: orderId
        })
        return data
    } catch (error) {
        console.error(error)
        return false
    }
}